src/main/webapp/static/js/init.js

```diff
@@ -1,10 +1,31 @@
 (function($){
   $(function(){
-
+    var web3Ready = false;
     $('.button-collapse').sideNav();
     $('#navButton').sideNav();
     
     $('select').material_select();
+    $('#web3').click(function() {
+      if (!web3Ready){
+        if (window.ethereum) {  
+          window.web3 = new Web3(ethereum);  
+          window.ethereum.enable()
+        } else if (window.web3) {
+          window.web3 = new Web3(web3.currentProvider)
+        } else {
+          console.log('No Metamask (or other Web3 Provider) installed')
+          return
+        }
+        web3Ready = true;
+        return
+      }
+      ethAddress = web3.currentProvider.selectedAddress;
+      console.log(ethAddress);
+      web3.eth.sign(ethAddress, 'sign-in',function(err, signature) {
```

Comment:
Is "sign-in" the message that will be visible to the user in the MetaMask window that pops up?

Reply:
Yeah, this message will be prompted to the user and asks to sign the message. we could change it to a more proper message!